support/k8s/k8s-workload-registrar/mode-reconcile/controllers/pod_controller.go

```diff
@@ -136,6 +142,10 @@ func (r *PodReconciler) getNamesForEndpoints(ctx context.Context, pod *corev1.Po
 
 	for _, endpoints := range endpointsList.Items {
 		endpoints := endpoints
+		if strings.Contains(endpoints.Name, ".") || strings.Contains(endpoints.Namespace, ".") {
```

Comment:
maybe we can make a small helper function to help with self documentation that returns true if it is likely a DNS name instead of doing the "contains dot" check everywhere?